src/InputBase.js

```diff
@@ -91,17 +92,20 @@ class InputBase extends React.Component {
   }
 
   renderIcon() {
-    let classes = {
-      'glyphicon': true,
-      'form-control-feedback': true,
-      'glyphicon-ok': this.props.bsStyle === 'success',
-      'glyphicon-warning-sign': this.props.bsStyle === 'warning',
-      'glyphicon-remove': this.props.bsStyle === 'error'
-    };
+    if (this.props.hasFeedback) {
+      if (this.props.feedbackIcon) {
+        return this.props.feedbackIcon;
```

Comment:
I am actually running into a similar issue with custom menus in the new dropdown stuff. One one hand forcing the user to reset the style if they want different behavior than `form-control-feedback` stinks, but I think that use case is less likely then someone just wanting to swap out the actual icon, in which case adding the class is the better option.